### PR TITLE
config changes

### PR DESCRIPTION
**Pull request by ishaansingh22**

### CODE REVIEW

File: src/configs/algo_config.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Algorithm Configuration
+
+iid_dispfl_clients_new = {
+    "algo": "dispfl",
+    "exp_id": 200,
+    "exp_type": "iid_dispfl",
+    "seed": 2,
```
Comment (tremblerz) [inline src/configs/algo_config.py:7]: let's move seed to system config

File: src/configs/sys_config.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# System Configuration
+
+system_config = {
+    "num_users": 4,
+    "experiment_path": "./experiments/",
+    "dset": "cifar10",
+    "dump_dir": "./expt_dump/",
+    "dpath": "/data/unagi0/anakewat/imgs/",
+    "device_ids": {
+        "iid_dispfl": {"node_0": [2], "node_1": [3], "node_2": [4], "node_3": [5], "node_4": [2]},
```
Comment (tremblerz) [inline src/configs/sys_config.py:11]: algorithm names shouldn't be present inside `device_ids` and system_config in general. The goal is to make the system config independent of algorithms.

File: src/configs/sys_config.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "iid_defkt": {"node_0": [4], "node_1": [4], "node_2": [4], "node_3": [4]},
+        "non_iid": {"node_0": [0], "node_1": [1], "node_2": [2], "node_3": [3]}
+    },
+    "dataset_splits": {
+        "iid": {"num_clients": 4, "samples_per_client": 500},
```
Comment (tremblerz) [inline src/configs/sys_config.py:16]: no need to mention multiple dataset split configs in a single config. It should looks something like following in my opinion -
```
"dataset_splits": {
  "type": "non-iid",
  "samples_per_user": 500
}
```

Comment (tremblerz) [reply src/configs/sys_config.py:16]: we should not have `num_clients` here because users is already specified above.

File: src/scheduler.py [hunk 4]
```diff
@@ -37,32 +40,47 @@
...
+        self.algo_config = load_config(algo_config_path)
+        self.merge_configs()
+
+    def merge_configs(self):
+        self.config = self.algo_config.copy()
+        self.config.update({
+            "dset": "cifar10",
+            "dump_dir": "./expt_dump/",
+            "dpath": self.sys_config["dataset_path"] + "cifar10",
+            "num_clients": self.sys_config["dataset_splits"]["iid"]["num_clients"],
```
Comment (tremblerz) [inline src/scheduler.py:79]: use `num_users`

**Review by tremblerz — CHANGES_REQUESTED** [review]

File: src/configs/sys_config.py (hunk 2 above)
Comment (ishaansingh22) [reply src/configs/sys_config.py:11]: how do we want to pass this in then

File: src/scheduler.py [hunk 5]
```diff
@@ -37,32 +39,47 @@
...
         self.config = process_config(self.config)
 
+    def assign_config_by_path(self, sys_config_path, algo_config_path):
+        self.sys_config = load_config(sys_config_path)
+        self.algo_config = load_config(algo_config_path)
+        self.merge_configs()
+
+    def merge_configs(self):
+        self.config = self.algo_config.copy()
+        self.config.update({
```
Comment (tremblerz) [inline src/scheduler.py:74]: Why is it hardcoded here?

File: src/configs/sys_config.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+system_config = {
+    "num_users": 4,
+    "experiment_path": "./experiments/",
+    "dset": "cifar10",
+    "dump_dir": "./expt_dump/",
+    "dpath": "/data/unagi0/anakewat/imgs/",
+    "seed": 2,
+    "device_ids": {}
+    "dataset_splits": {
+        "iid": {"samples_per_user": 500},
```
Comment (tremblerz) [inline src/configs/sys_config.py:11]: only one dataset split should be present in a single configuration. You can make two separate configuration if you want to represent both configurations.

**Review by tremblerz — CHANGES_REQUESTED** [review]

File: src/scheduler.py [hunk 7]
```diff
@@ -37,32 +39,47 @@
...
+            "dset": "cifar10",
```
Comment (tremblerz) [inline src/scheduler.py:75]: can you remove this hardcoding? We do not want any parameter being hardcoded inside scheduler.py
Only the config needs to be the single source of truth.

**Review by tremblerz — CHANGES_REQUESTED** [review]

**Review by tremblerz — APPROVED** [review]

**Pull request merged**